pkg/virtual/workspaces/auth/cache.go

```diff
@@ -181,6 +181,9 @@ type AuthorizationCache struct {
 	syncHandler func(request *reviewRequest, userSubjectRecordStore cache.Store, groupSubjectRecordStore cache.Store, reviewRecordStore cache.Store) error
 
 	rwMutex sync.RWMutex
+
+	watcherLock sync.Mutex
```

Comment:
rwMutex and watcherLock? Why two?

Reply:
well, it just doesn't protect the same thing :-)

Reply:
What does rwMutex protect? It's not clear from the new call-sites of rwMutex.RLock